src/mlpack/core/optimizers/saga/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+set(SOURCES
+        saga.hpp
+        saga_impl.hpp
+        saga_update.hpp
+        )
```

Comment:
This looks strange, can you take a look at the other ``CMakeLists.txt`` files and make some adjustments.